### PR TITLE
Add GenericSystem to GPIO example

### PR DESCRIPTION
**Pull request by christophfroehlich**
A test for reproducing https://github.com/ros-controls/ros2_control/issues/1277 but I think this is valuable also to add here.

### CODE REVIEW

File: example_10/doc/userdoc.rst [hunk 1]
```diff
@@ -91,6 +91,96 @@ The *RRBot* URDF files can be found in the ``description/urdf`` folder.
...
+    Hardware Component 1
+        name: RRBot
+        type: system
+        plugin name: ros2_control_demo_example_10/RRBotSystemWithGPIOHardware
+        state: id=3 label=active
+        command interfaces
+                joint1/position [available] [claimed]
+                joint2/position [available] [claimed]
+                flange_analog_IOs/analog_output1 [available] [claimed]
+                flange_vacuum/vacuum [available] [claimed]
```
Comment (bailaC) [inline example_10/doc/userdoc.rst:113]: One general comment, should we indicate that flange_vacuum is the gpio mock hardware?

Comment (christophfroehlich) [reply example_10/doc/userdoc.rst:113]: I pushed an update, what do you think now?

Comment (bailaC) [reply example_10/doc/userdoc.rst:113]: looks good. thanks for considering.

**Review by bailaC — APPROVED** [review]

**Review by progtologist — APPROVED** [review]

**Review by saikishor — APPROVED** [review]
LGTM

**Pull request merged**